### PR TITLE
Chrom coverage as json

### PR DESCRIPTION
**Pull request by northwestwitch**
### This PR adds | fixes:
- Adds an endpoint that calculates the mean coverage over all transcripts of an entire chromosome for one or more samples and returns it as json.

- Fixes an error on the endpoint that collects the coverage over genes and returns a json object.

**How to prepare for test**:
- [x] from user hiseq.clinical, install the branch on clinical-db stage, install the branch mt_vs_automosome_coverage of scout:
```
cd home/hiseq.clinical/servers/resources/clinical-db.scilifelab.se
bash update-scout-stage.sh mt_vs_automosome_coverage
```
- [x] Then install this chanjo-report branch on clinical-db stage:
`bash update-chanjo-report-stage.sh chrom_coverage_as_json`

#### Test number 1: get chromosome coverage for all samples of a case. Example: chromosome 21 and samples from this case: https://scout-stage.scilifelab.se/cust000/643594-300M
- [x] From a terminal window (clinical-db stage environment) run the curl POST request:
```
curl -i -H "Content-Type: application/json" -X POST \
-d '{"sample_ids":"ADM1059A4,ADM1059A5,ADM1059A6", "chrom": "21"}' \
https://scout-stage.scilifelab.se/reports/json_chrom_coverage
```

#### Test number 2. Fix to the gene coverage endpoint -> Get the coverage for samples of the same case over one gene (example POT1 hgnc id: 17284) using this request from same terminal:

```
curl -i -H "Content-Type: application/json" -X POST \
-d '{"sample_ids":"ADM1059A4,ADM1059A5,ADM1059A6", "gene_ids": "17284"}' \
https://scout-stage.scilifelab.se/reports/json_gene_coverage
```

and compare to value from scout coverage report.

### Expected outcome:
- [x] test 1: A json object with sample coverage over chr 21 should be returned
- [x] test 2: Gene coverage returned by json object should be the same as the one in scout coverage report

### Review:
- [ ] Code approved by
- [x] Tests executed by CR
- [ ] "Merge and deploy" approved by

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Comment by northwestwitch** [comment]
- [x] Test n.1 : get mean coverage over chromosome 21 for samples of case 643594-300M : 🆗 

image

- [x] Test n.2 : Gene coverage over POT 1 gene is the same in the json object and scout coverage report: 🆗 

image

image

**Comment by northwestwitch** [comment]
Another small one perhaps @moonso ?

File: chanjo_report/server/blueprints/report/views.py [hunk 1]
```diff
@@ -63,6 +63,28 @@ def genes():
...
 
+
+@report_bp.route('/json_chrom_coverage', methods=['POST'])
+def json_chrom_coverage():
+    """Calculate mean coverage over all gene transcripts of a chromosome for one or more samples and return results as json"""
+
+    results = {}
+    data = request.json
+
+    if data.get('chrom') is None or data.get('sample_ids') is None:
```
Comment (moonso) [inline chanjo_report/server/blueprints/report/views.py:74]:
```suggestion
    if not (data.get('chrom') and data.get('sample_ids')):
```

**Review by moonso — APPROVED** [review]
Same comment as yesterday 😄

File: chanjo_report/server/blueprints/report/views.py (hunk 1 above)
Comment (northwestwitch) [reply chanjo_report/server/blueprints/report/views.py:74]: Whoops! Thanks! 😄

**Pull request merged**